sbi/analysis/plot.py

```diff
@@ -1403,3 +1408,166 @@ def _plot_hist_region_expected_under_uniformity(
         color=color,
         alpha=alpha,
     )
+
+
+# ==== Diagnostics for L-C2ST ====
+
+
+def pp_plot_lc2st(
+    probas,
+    probas_null,
+    labels,
+    colors,
+    conf_alpha=0.05,
+    pp_vals_null=None,
+    ax=None,
+    **kwargs,
+):
+    """Probability - Probability (P-P) plot for the classifier predicted
+    class probabilities in (L)C2ST to assess the validity of a (or several)
+    density estimator(s).
+
+    Args:
+        probas (list of np.arrays): list of predicted class probabilities for each case
+            (i.e. associated to each density estimator).
+        probas_null (list of ): list of predicted class probabilities in each trial
+            under the null hypothesis.
+        labels (list): list of labels for every density estimator.
+        colors (list): list of colors for every density estimator.
+        pp_vals_null (dict): dictionary of PP-values for each trial under the null
+            hypothesis.
+        ax (matplotlib.axes.Axes): axes to plot on.
+        **kwargs: keyword arguments for matplotlib.pyplot.plot.
+
+    Returns:
+        ax (matplotlib.axes.Axes): axes with the P-P plot.
+    """
+    if ax is None:
+        ax = plt.gca()
+    alphas = np.linspace(0, 1, 100)
+    pp_vals_dirac = PP_vals([0.5] * len(probas), alphas)
+    ax.plot(alphas, pp_vals_dirac, "--", color="black", label="True probability (H0)")
+
+    if pp_vals_null is None:
+        pp_vals_null = {}
+        for t in range(len(probas_null)):
+            pp_vals_null[t] = pd.Series(PP_vals(probas_null[t], alphas))
+
+    low_null = pd.DataFrame(pp_vals_null).quantile(conf_alpha / 2, axis=1)
+    up_null = pd.DataFrame(pp_vals_null).quantile(1 - conf_alpha / 2, axis=1)
```

Comment:
maybe you don't need pandas here? you can use https://numpy.org/doc/stable/reference/generated/numpy.quantile.html
I don't know if pandas is already a dependency of sbi but it's always a good idea to limit dependencies when they can be avoided (think import time, CI issues on updates)

Reply:
I'm fairly certain that pandas it is a dependency, but I agree. The small fixes you suggested are totally doable. However I would also have to replace the `.groupy` method from pandas, which will probably require to implement a `numpy_groupy` function...

Reply:
you can use https://docs.scipy.org/doc/scipy/reference/generated/scipy.stats.binned_statistic.html#

but now if Pandas is already used everywhere I think it's fair to ignore my point...